src/ModelContextProtocol.AspNetCore/IdleTrackingBackgroundService.cs

```diff
@@ -56,26 +61,34 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                         continue;
                     }
 
-                    idleSessions.Add((session.LastActivityTicks, session.Id));
+                    // Add the timestamp and the session
+                    idleSessionsTimestamps.Add(session.LastActivityTicks);
+                    idleSessionSessionIds.Add(session.Id);
 
                     // Emit critical log at most once every 5 seconds the idle count it exceeded,
                     // since the IdleTimeout will no longer be respected.
-                    if (idleSessions.Count == maxIdleSessionCount + 1)
+                    if (idleSessionsTimestamps.Count == maxIdleSessionCount + 1)
                     {
                         LogMaxSessionIdleCountExceeded(maxIdleSessionCount);
                     }
                 }
 
-                if (idleSessions.Count > maxIdleSessionCount)
+                if (idleSessionsTimestamps.Count > maxIdleSessionCount)
                 {
-                    var sessionsToPrune = idleSessions.ToArray()[..^maxIdleSessionCount];
-                    foreach (var (_, id) in sessionsToPrune)
+                    var timestamps = CollectionsMarshal.AsSpan(idleSessionsTimestamps);
+
+                    // Sort only if the maximum is breached and sort solely by the timestamp. Sort both collections.
+                    timestamps.Sort(CollectionsMarshal.AsSpan(idleSessionSessionIds));
```

Comment:
I wasn't aware this version of Sort existed. Pretty cool. However, wouldn't it be easier to use `List<(long Timestamp, string SessionId)>` and sort that though? Then we don't need to use CollectionsMarshal either which I see as a win. Even though the usage here is safe, I prefer to use less esoteric APIs if possible.